SQL_Assignment_3.sql

```diff
@@ -0,0 +1,258 @@
+SELECT CONVERT(DATE, GETDATE()) AS CURRENT_DATE;
+--34. From an 'Orders' table with an 'OrderTotal' column, display the total as a currency with 2 decimal places.
+SELECT CAST(SUM(OrderTotal) AS decimal(10,2) ) AS ODERTOTAL_FORMATED;
+
+--35. Separate the string 'apple,banana,cherry' into individual fruits.
+SELECT VALUE AS FRUIT FROM
+STRING_SPLIT('apple,banana,cherry',',') AS FRUITS
+--36. From a 'Skills' table with a 'SkillList' column containing comma-separated skills, write a query to create a row 
+--for each individual skill.
+
+SELECT SKILL_LIST FROM Skills
+STRING_SPLIT(SkillList,',') AS SKILL_LIST;
+-----------------------------------------------------------------------------------------
+
+--Date and Time Functions
+
+--37. Write a query to display the current date and time.
+SELECT GETDATE()
+
+--38. From an 'Orders' table, find all orders placed in the last 24 hours.
+SELECT *
+FROM ODERS
+WHERE ODERDATE>=DATEADD(day, -1, GETDATE());
+
+--39. Display the current UTC date and time.
+SELECT GETUTCDATE() AS CurrentUTCDateTime;
+
+--40. Write a query to show the time difference between local time and UTC time.
+SELECT 
+    SYSDATETIMEOFFSET() AS LocalDateTimeWithOffset,
+    GETUTCDATE() AS UTCDateTime,
+    DATEDIFF(MINUTE, GETUTCDATE(), SYSDATETIMEOFFSET()) AS DifferenceInMinutes
+
+--41. Convert the current date and time to 'Pacific Standard Time'.
+SELECT 
+    SYSDATETIME() AS CurrentDateTime,
+    SYSDATETIME() AT TIME ZONE 'UTC' AT TIME ZONE 'Pacific Standard Time' AS PSTDateTime;
+--42. From a 'Flights' table with a 'DepartureTime' column in UTC, convert all departure times to 'Eastern Standard Time'.
+SELECT DepartureTime  AT TIME ZONE 'UTC' AT TIME ZONE 'Eastern Standard Time' AS EASTERN_STD_TIME
+FROM FLIGHTS ;
+
+
+--43. Add 3 months to the current date.
+   SELECT DATEADD(MONTH, 3, GETDATE()) AS NewDate;
+
+--44. From an 'Employees' table, write a query to calculate each employee's retirement date (65 years from their 'DateOfBirth').
+SELECT DATEADD(YEAR, 65, DateOfBirth) AS RETIERMENT
+FROM  Employees;
+
+--45. Calculate the number of days between '2023-01-01' and '2023-12-31'.
+SELECT DATEDIFF(DAY, '2023-01-01' , '2023-12-31') AS DAYBETWEEN;
+
+--46. From an 'Orders' table, find the average number of days between order date and ship date.
+select * from Orders
+SELECT AVG(DATEDIFF(DAY,SHIPDATE, OrderDate)) AS AverageDateBetween FROM Orders;
+
+--47. Extract the month number from the date '2023-09-15'.
+SELECT MONTH('2023-09-15') AS MONTH;
+
+--48. From a 'Sales' table, write a query to group total sales by the quarter of the sale date.
+SELECT DATEPART(QUARTER, SALEDATE) AS Quarter,SUM(SALEAMOUNT) as SalesInQuarter
+FROM SALES
+GROUP BY DATEPART(QUARTER,SALEDATE);
+
+--49. Extract the year from the current date.
+SELECT DATEPART(YEAR, GETDATE()) AS YEAR;
+
+--50. From an 'Employees' table, find all employees hired in the year 2022.
+WHERE DATEPART(YEAR, HIREDATE)='2020';
```

Comment:
Missing FROM clause and the year condition is incorrect.

Reply:
changes made